merkle_zeppelin/trees/merkle.py

```diff
@@ -16,67 +20,141 @@ def keccak256(v: bytes) -> bytes:
 class MerkleTree(BinaryTree[bytes]):
     def __init__(
         self,
-        leaves: list[Leaf],
+        raw_elements: list[Leaf],
         types: list[str],
         hashing_function: Callable[[bytes], bytes] = None,
     ) -> None:
         self._hashing_function = hashing_function or keccak256
         self._types = types
 
-        hash_leaf_pairs = self._get_hash_leaf_pairs(leaves)
-        hash_leaf_pairs.sort(reverse=True)
+        sorted_index_hash_pairs = self._get_sorted_index_hash_pairs(raw_elements)
+        sorted_hashes = [hash_ for index, hash_ in sorted_index_hash_pairs]
+
+        super().__init__(sorted_hashes)
 
-        ordered_hashed_leaves = [leaf_hash for leaf_hash, _ in hash_leaf_pairs]
-        ordered_leaves = [leaf for _, leaf in hash_leaf_pairs]
+        self._raw_to_leaves_index_mapping = self._get_raw_to_leaves_index_mapping(
+            raw_elements, sorted_index_hash_pairs
+        )
 
-        super().__init__(ordered_hashed_leaves)
+    @staticmethod
+    def get_hash_from_string(hash_: str) -> bytes:
+        if hash_.startswith("0x"):
+            hash_ = hash_[2:]
 
-        self._raw_leaves_index = self._get_raw_leaves_index(ordered_leaves)
+        return bytes.fromhex(hash_)
+
+    @classmethod
+    def import_tree(
+        cls,
+        data: Any,
+        importer: Type[MerkleTreeImporter],
+        validate: bool = True,
+        hashing_function: Callable[[bytes], bytes] = None,
+    ) -> MerkleTree:
+        imported_data = importer.import_tree(data)
+        obj = cls.__new__(cls)
+        obj._hashing_function = hashing_function or keccak256
+        obj._nodes = [
+            cls.get_hash_from_string(hash_str) for hash_str in imported_data.tree
+        ]
+        obj._types = imported_data.leaf_encoding
+        obj._raw_to_leaves_index_mapping = {
+            value.value: value.tree_index for value in imported_data.values
+        }
+
+        if validate:
+            cls.validate(obj)
+
+        return obj
+
+    @property
+    def dto(self) -> MerkleTreeDTO:
+        return MerkleTreeDTO(
+            tree=[f"0x{node.hex()}" for node in self._nodes],
+            values=[
+                LeafValueDTO(value=value, treeIndex=index)
+                for value, index in self._raw_to_leaves_index_mapping.items()
+            ],
+            leafEncoding=self._types,
```

Comment:
Why it is camelcase?

Reply:
so unfortunately I was not able to make import and export to json with camelcase and also keep initialization with snake case
I guess I will add issue or even new PR without issue fixing this

Reply:
Could you share some details?